### ISSUE TITLE
improve pre-commit hook

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/PermaplanT#159]
- The current pre-commit hook is too slow. Furthermore, it does not do fast and essential things like reformatting.
- Reformatting, however, should be done after Milestone 1 (#35), to not postpone the Milestone.
- Backend Linting should be checked if fast enough.
